repositories/strategies/routingStrategy.js

```diff
@@ -0,0 +1,287 @@
+  trx("PagesView")
+    .where({ id })
+    .first()
+    .then(fromDb);
+
+const getSectionById = (trx, id) =>
+  trx("Sections")
+    .where({ id, isDeleted: false })
+    .first()
+    .then(fromDb);
+
+const getPageDestinations = (trx, { sectionId, order }) =>
+  trx("PagesView")
+    .where(toDb({ sectionId }))
+    .where("order", ">", order)
+    .then(map(fromDb));
+
+const getSectionDestinations = (trx, { id, questionnaireId }) =>
+  trx("Sections")
+    .select("*")
+    .where(toDb({ questionnaireId, isDeleted: false }))
+    .where("id", ">", id)
+    .then(map(fromDb));
+
+const findRoutingRuleSet = (trx, questionPageId) =>
+  trx("Routing_RuleSets")
+    .where(toDb({ questionPageId, isDeleted: false }))
+    .first()
+    .then(fromDb);
+
+const getRoutingRuleSetById = (trx, routingRuleSetId) => {
+  return trx("Routing_RuleSets")
+    .where(toDb({ id: routingRuleSetId, isDeleted: false }))
+    .first()
+    .then(fromDb);
+};
+
+const insertRoutingCondition = (trx, routingCondition) =>
+  trx("Routing_Conditions")
+    .insert(toDb(routingCondition))
+    .returning("*")
+    .then(head)
+    .then(fromDb);
+
+const insertRoutingRule = (trx, routingRule) =>
+  trx("Routing_Rules")
+    .insert(toDb(routingRule))
+    .returning("*")
+    .then(head)
+    .then(fromDb);
+
+const insertRoutingRuleSet = async (trx, questionPageId) =>
+  trx("Routing_RuleSets")
+    .insert(
+      toDb({
+        questionPageId: parseInt(questionPageId)
+      })
+    )
+    .returning("*")
+    .then(head)
+    .then(fromDb);
+
+const checkAnswerBelongsToPage = async (trx, answerId, questionPageId) => {
+  if (isNil(answerId)) {
+    return;
+  }
+
+  const answer = await findAnswerByIdAndQuestionPageId(
+    trx,
+    answerId,
+    questionPageId
+  );
+  if (!answer) {
+    throw new Error(
+      `Answer ${answerId} does not belong to ${questionPageId}. Choose a different answer.`
+    );
+  }
+};
+
+const getAnswerOrFirstAnswerOnPage = async (trx, answerId, questionPageId) => {
+  if (!isNil(answerId)) {
+    return answerId;
+  }
+  const firstAnswer = await getFirstAnswer(trx, questionPageId);
+  return get(firstAnswer, "id", null);
+};
+
+const updateRoutingConditionStrategy = async (
+  trx,
+  { id: routingConditionId, questionPageId, answerId }
+) => {
+  checkAnswerBelongsToPage(trx, answerId, questionPageId);
+  const routingConditionAnswer = await getAnswerOrFirstAnswerOnPage(
+    trx,
+    answerId,
+    questionPageId
+  );
+  await deleteRoutingConditionValues(trx, routingConditionId);
+  return updateRoutingCondition(
+    trx,
+    routingConditionId,
+    questionPageId,
+    routingConditionAnswer
+  );
+};
+
+const toggleConditionValueStrategy = async (
+  trx,
+  { conditionId, optionId, checked }
+) => {
+  const table = trx("Routing_ConditionValues");
+  const args = toDb({ conditionId, optionId });
+  const existing = await table.where(args);
+  const shouldInsert = checked || isEmpty(existing);
```

Comment:
Should probably throw an error here if `!isEmpty(existing)&&checked` then the query can just check checked to see if its an insert or delete query.